corehq/motech/repeaters/management/commands/create_missing_repeat_records.py

```diff
@@ -0,0 +1,213 @@
+                    'percentage_missing': f'{round((total_missing_count / total_count) * 100, 2)}%',
+                }
+            }
+
+    return stats_per_domain
+
+
+def obtain_missing_form_repeat_records_in_domain(domain, repeaters, form, should_create):
+    if form.is_duplicate:
+        return 0, 0
+
+    missing_count = 0
+    successful_count = 0
+    repeat_records = get_repeat_records_by_payload_id(domain, form.get_id)
+    triggered_repeater_ids = [record.repeater_id for record in repeat_records]
+    for repeater in repeaters:
+        if not repeater.allowed_to_forward(form):
+            continue
+
+        if repeater.get_id in triggered_repeater_ids:
+            successful_count += 1
+        else:
+            missing_count += 1
+            if should_create:
+                # will attempt to send now
+                repeater.register(form)
+
+    return missing_count, successful_count
+
+
+def get_form_ids_in_domain_between_dates(domain, startdate, enddate):
+    return FormES(es_instance_alias='export').domain(domain)\
+        .date_range('server_modified_on', gte=startdate, lte=enddate).source(['_id']).run().hits
+
+
+def obtain_missing_case_repeat_records(startdate, enddate, domains):
+    stats_per_domain = {}
+    for domain in domains:
+        total_missing_count = 0
+        total_count = 0
+        case_repeaters_in_domain = get_case_repeaters_in_domain(domain)
+
+        case_ids = [c['_id'] for c in get_case_ids_in_domain_since_date(domain, startdate)]
+        cases = CaseAccessors(domain).get_cases(case_ids)
+        for case in cases:
+            missing_count, successful_count = obtain_missing_case_repeat_records_in_domain(
+                domain, case_repeaters_in_domain, case, startdate, enddate
+            )
+            total_missing_count += missing_count
+            total_count += missing_count + successful_count
+
+        if total_missing_count > 0:
+            stats_per_domain[domain] = {
+                'cases': {
+                    'missing_count': total_missing_count,
+                    'percentage_missing': f'{round((total_missing_count / total_count) * 100, 2)}%',
+                }
+            }
+
+    return stats_per_domain
+
+
+def obtain_missing_case_repeat_records_in_domain(domain, repeaters, case, startdate, enddate):
+    successful_count = 0
+    missing_count = 0
+
+    repeat_records = get_repeat_records_by_payload_id(domain, case.get_id)
+    records_in_daterange = [record for record in repeat_records
+                            if startdate <= record.registered_on.date() <= enddate]
+
+    triggered_repeater_ids_and_counts = {}
+    for record in records_in_daterange:
+        current_count = triggered_repeater_ids_and_counts.get(record.repeater_id, 0)
+        if current_count > 0:
+            triggered_repeater_ids_and_counts[record.repeater_id] += 1
+        else:
+            triggered_repeater_ids_and_counts[record.repeater_id] = 1
```

Comment:
nit: perfect use case for defaultdict
```suggestion
    triggered_repeater_ids_and_counts = defaultdict(int)
    for record in records_in_daterange:
        triggered_repeater_ids_and_counts[record.repeater_id] += 1
```

Another way you can do this is with a `Counter` (which is a subclass of dict):
```python
from collections import Counter
triggered_repeater_ids_and_counts = Counter([record.record_id for record in records_in_daterange])
```

Reply:
Thank you! Makes sense.